src/main/kotlin/stringAddCalculator/StringAddCalculatorInput.kt

```diff
@@ -0,0 +1,42 @@
+package stringAddCalculator
+
+import java.lang.RuntimeException
+
+class StringAddCalculatorInput(private val input: String) {
+    fun parse(): List<Int> {
+        val customDelimiter = getCustomDelimiter()
+
+        return getIntegerPart().split(Regex("[${DEFAULT_DELIMITERS.joinToString()}$customDelimiter]"))
+            .map {
+                try {
+                    convertToInt(it)
+                } catch (e: NumberFormatException) {
+                    throw RuntimeException()
+                }
+            }
+    }
+
+    private fun getCustomDelimiter(): String {
+        val result = Regex("//(.)\n(.*)").find(input) ?: return ""
+
+        return result.groupValues[1]
+    }
+
+    private fun getIntegerPart(): String = input.split("\n").last()
+
+    private fun convertToInt(text: String): Int {
+        try {
+            return text.toInt().also {
+                if (it < 0) {
+                    throw RuntimeException()
+                }
+            }
+        } catch (e: NumberFormatException) {
+            throw RuntimeException()
+        }
+    }
```

Comment:
p2; try-catch 비용은 싸지 않습니다. 다음과 같이 리펙토링 해 볼 수도 있을 거 같아요.

```suggestion
    private fun convertToInt(text: String): Int = text.toIntOrNull()
        ?.takeIf { it >= ZERO }
        ?: throw RuntimeException()
```

Reply:
`takeIf`라는 함수가 있군요..! 제가 찾고있던 함수인 것 같습니다. 감사합니다!

제이슨님이 해주신 강의에서도 그렇고 위에 해주신 RuntimeException 관련 피드백과 마찬가지로
여러 환경에서 코틀린 코드가 작동할 것을 고려해서 java 패키지로부터 import하는 것을 꼭 필요한 상황이 아니라면 지양하시는 것으로 느꼈습니다.

그래서 ZERO는 제가 확인할 때는 java 패키지로부터 불러오는 것 같아서 동반객체로 선언하고 0에 의미를 부여해보았습니다!

Reply:
맞습니다. 자바의 패키지 의존성을 줄이는 것이 코틀린스럽게 사용하는 것이기도 하다고 느끼고 있어요.
ZERO는 java 패키지의 const가 아닌 커스텀 const를 의도한 거기도 합니다. 😊